app/models/miq_queue.rb

```diff
@@ -641,6 +639,29 @@ def self.optional_values(options, keys = [:zone])
 
   private_class_method :optional_values
 
+  def self.messaging_client_options
+    queue_settings = Settings.prototype[queue_type]
+
+    {
+      :host     => ENV["QUEUE_HOSTNAME"] || queue_settings.queue_hostname,
+      :port     => (ENV["QUEUE_PORT"] || queue_settings.queue_port).to_i,
+      :username => ENV["QUEUE_USERNAME"] || queue_settings.queue_username,
+      :password => ENV["QUEUE_PASSWORD"] || queue_settings.queue_password,
+      :protocol => messaging_client_protocol
+    }
+  end
+  private_class_method :messaging_client_options
+
+  def self.messaging_client_protocol
+    case queue_type
+    when "artemis"
+      :Stomp
+    when "kafka"
+      :Kafka
+    end
+  end
+  private_class_method :messaging_client_protocol
```

Comment:
As discussed, I feel like this should be pushed down into manageiq-messaging... feels like something the caller shouldn't have to know about.

Reply:
Yeah I agree